pkg/controller/publisher.go

```diff
@@ -167,29 +167,33 @@ func (s *SCSIPublisher) addTargetsInfoToPublishContext(
 	publishContext map[string]string, volumeApplianceID string, client gopowerstore.Client) error {
 	iscsiTargetsInfo, err := common.GetISCSITargetsInfoFromStorage(client, volumeApplianceID)
 	if err != nil {
-		return err
+		log.Error("error unable to get iSCSI targets from array", err)
 	}
 	for i, t := range iscsiTargetsInfo {
 		publishContext[fmt.Sprintf("%s%d", common.PublishContextISCSIPortalsPrefix, i)] = t.Portal
 		publishContext[fmt.Sprintf("%s%d", common.PublishContextISCSITargetsPrefix, i)] = t.Target
 	}
 	fcTargetsInfo, err := common.GetFCTargetsInfoFromStorage(client, volumeApplianceID)
 	if err != nil {
-		return err
+		log.Error("error unable to get FC targets from array", err)
 	}
 	for i, t := range fcTargetsInfo {
 		publishContext[fmt.Sprintf("%s%d", common.PublishContextFCWWPNPrefix, i)] = t.WWPN
 	}
 
+	// There is no API availble for NVMeTCP and hence targets are added in node staging using goNVMe
 	nvmefcTargetInfo, err := common.GetNVMEFCTargetInfoFromStorage(client, volumeApplianceID)
 	if err != nil {
-		return err
+		log.Error("error unable to get NVMeFC targets from array", err)
 	}
 	for i, t := range nvmefcTargetInfo {
 		publishContext[fmt.Sprintf("%s%d", common.PublishContextNVMEFCPortalsPrefix, i)] = t.Portal
 		publishContext[fmt.Sprintf("%s%d", common.PublishContextNVMEFCTargetsPrefix, i)] = t.Target
 	}
-
+	// If the system is not capable of any protocol, then we will through the error
+	if len(iscsiTargetsInfo) == 0 && len(fcTargetsInfo) == 0 && len(nvmefcTargetInfo) == 0 {
```

Comment:
do we need to add check for nvme TCP targets also?

Reply:
I don't see any specific target information for NVMeTCP @harshitap26 please confirm whether we need it separately?

Reply:
Can we remove this condition from here and add it in the nodeStageVolume function after we get the NVMeTCP targets as well? @harishp8889

Reply:
ok